core/store/src/main/java/mil/nga/giat/geowave/core/store/DataStore.java

```diff
@@ -267,6 +268,29 @@ public boolean deleteEntry(
 
 	/**
 	 * Returns all data in this data store that matches the query parameter
+	 * within the index described by the index passed in. All data types that
+	 * match the query will be returned as an instance of the native data type
+	 * that was originally ingested, but only the subset of attributes specified
+	 * will be included.
+	 * 
+	 * @param index
+	 *            The index information to query against. All data within the
+	 *            index of this index ID will be queried and returned.
+	 * @param attributes
+	 *            The subset of attributes to include in the results
+	 * @param query
+	 *            The description of the query to be performed
+	 * @return An iterator on all results that match the query. The iterator
+	 *         implements Closeable and it is best practice to close the
+	 *         iterator after it is no longer needed.
+	 */
+	public <T> CloseableIterator<T> query(
```

Comment:
arguably, we should start adding things like this subsetting, as well as possibly the limit and maybe other things to be contained in the query object so we limit the number of methods in DataStore.

also, "attributes" are specific to simplefeature data, I think more generically you are filtering the "fieldIds"